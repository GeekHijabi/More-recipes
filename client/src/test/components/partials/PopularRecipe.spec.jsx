import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ConnectedPopularRecipe, { PopularRecipe }
  from '../../../js/components/Partials/PopularRecipe';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);


const props = {
  apiGetAllFavoriteRecipes: jest.fn(() => Promise.resolve({ ok: true })),
  // searchItem: jest.fn(() => Promise.resolve({ ok: true }))
};
describe('Component: PopularRecipe', () => {
  beforeEach(() => {
    global.toastr = {
      success: () => {},
      error: () => {}
    };
  });

  describe('PopularRecipe in snapshot', () => {
    it('should render correctly', () => {
      const wrapper = shallow(<PopularRecipe {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Connected PopularRecipe component', () => {
    it('component successfully rendered', () => {
      const store = mockStore({
        recipe: {
          favRecipes: []
        }
      });
      const wrapper = shallow(<ConnectedPopularRecipe store={store} />);
      expect(wrapper.length).toBe(1);
    });
  });
});

