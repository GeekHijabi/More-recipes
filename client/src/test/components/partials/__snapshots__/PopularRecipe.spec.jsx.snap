// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PopularRecipe PopularRecipe in snapshot should render correctly 1`] = `<div />`;
